File: app/src/main/java/mb00/android/codehub/ui/repo/viewmodel/RepoCommitsViewModel.java
package mb00.android.codehub.ui.repo.viewmodel;

import mb00.android.codehub.ui.repo.interactor.RepoInteractor;
import mb00.android.codehub.ui.repo.router.RepoRouter;


public class RepoCommitsViewModel extends RepoViewModel {

    public RepoCommitsViewModel(RepoInteractor interactor, RepoRouter router) {
        super(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/repo/viewmodel/RepoCommitsViewModel.kt
package mb00.android.codehub.ui.repo.viewmodel

import io.reactivex.Single
import mb00.android.codehub.api.model.Commit
import mb00.android.codehub.ui.repo.interactor.RepoInteractor
import mb00.android.codehub.ui.repo.router.RepoRouter


class RepoCommitsViewModel(private val interactor: RepoInteractor, private val router: RepoRouter)
    : RepoViewModel(interactor, router) {

    fun loadRepoCommits(header: String, user: String, repo: String): Single<List<Commit>> {
        return interactor.loadRepoCommits(header, user, repo)
    }

}
